service/src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceServiceTest.java

```diff
@@ -1968,28 +1981,33 @@ public void updateControlledBigQueryDataset_HandlesDatasets_WithOneNullLifetime(
     ControlledBigQueryDatasetResource createdDataset =
         makeBigQueryDatasetWithLifetime(
             /*defaultTableLifetime=*/ null, DEFAULT_CREATED_BIG_QUERY_PARTITION_LIFETIME);
+    try {
+      // Check which BQ datasets' lifetime to update.
+      List<ControlledBigQueryDatasetResource> emptyList =
+          updateControlledBigQueryDatasetsLifetimeAndWait();
 
-    // Check which BQ datasets' lifetime to update.
-    List<ControlledBigQueryDatasetResource> emptyList =
-        updateControlledBigQueryDatasetsLifetimeAndWait();
-
-    // Update nothing because one of the lifetimes is populated (the SQL query checks for AND - both
-    // lifetimes null).
-    assertTrue(emptyList.isEmpty());
+      // Update nothing because one of the lifetimes is populated (the SQL query checks for AND -
+      // both lifetimes null).
+      assertTrue(emptyList.isEmpty());
 
-    // Artificially set lifetimes to null in the database.
-    resourceDao.updateBigQueryDatasetDefaultTableAndPartitionLifetime(createdDataset, null, null);
+      // Artificially set lifetimes to null in the database.
+      resourceDao.updateBigQueryDatasetDefaultTableAndPartitionLifetime(createdDataset, null, null);
 
-    List<ControlledBigQueryDatasetResource> updatedResourceList =
-        updateControlledBigQueryDatasetsLifetimeAndWait();
+      List<ControlledBigQueryDatasetResource> updatedResourceList =
+          updateControlledBigQueryDatasetsLifetimeAndWait();
 
-    // The controlled dataset is updated since the lifetime is null.
-    assertEquals(1, updatedResourceList.size());
-    assertControlledBigQueryDatasetLifetimeIsUpdatedAndActivityIsLogged(
-        updatedResourceList,
-        createdDataset.getResourceId(),
-        /*expectedTableLifetime=*/ null,
-        DEFAULT_CREATED_BIG_QUERY_PARTITION_LIFETIME);
+      // The controlled dataset is updated since the lifetime is null.
+      assertEquals(1, updatedResourceList.size());
+      assertControlledBigQueryDatasetLifetimeIsUpdatedAndActivityIsLogged(
+          updatedResourceList,
+          createdDataset.getResourceId(),
+          /*expectedTableLifetime=*/ null,
+          DEFAULT_CREATED_BIG_QUERY_PARTITION_LIFETIME);
+    } finally {
+      // Remove dataset to not conflict with other test that checks for empty lifetime
+      controlledResourceService.deleteControlledResourceSync(
+          workspaceId, createdDataset.getResourceId(), userAccessUtils.defaultUserAuthRequest());
+    }
   }
```

Comment:
Surrounding this test with a `try... finally` clause so the BQ dataset is always removed.